test/Atmos/EDMF/report_mse_bomex.jl

```diff
@@ -33,6 +33,24 @@ function test_mse(computed_mse, best_mse, key)
     mse_not_regressed || @show key
 end
 
+#! format: off
+dons(cache_vs_z) = Dict(
+    "z" => [ca.z for ca in cache_vs_z],
+    "ρ" => [ca.prog.ρ for ca in cache_vs_z],
+    "ρu[1]" => [ca.prog.ρu[1] for ca in cache_vs_z],
+    "moisture.ρq_tot" => [ca.prog.moisture.ρq_tot for ca in cache_vs_z],
+    "turbconv.updraft[1].ρa" => [ca.prog.turbconv.updraft[1].ρa for ca in cache_vs_z],
+    "turbconv.updraft[1].ρaw" => [ca.prog.turbconv.updraft[1].ρaw for ca in cache_vs_z],
+    "turbconv.updraft[1].ρaθ_liq" => [ca.prog.turbconv.updraft[1].ρaθ_liq for ca in cache_vs_z],
+    "turbconv.updraft[1].ρaq_tot" => [ca.prog.turbconv.updraft[1].ρaq_tot for ca in cache_vs_z],
+    "turbconv.environment.ρatke" => [ca.prog.turbconv.environment.ρatke for ca in cache_vs_z],
+    "turbconv.environment.ρaθ_liq_cv" => [ca.prog.turbconv.environment.ρaθ_liq_cv for ca in cache_vs_z],
+    "turbconv.environment.ρaq_tot_cv" => [ca.prog.turbconv.environment.ρaq_tot_cv for ca in cache_vs_z],
```

Comment:
This is performing 11 passes through `cache_vs_z` when it could use just 1 with 11 `push!`s...

Reply:
Hmm, it looks like the dynamic allocations from `push!` are more costly:
```julia
julia> # let

       #! format: off
       function dons_single_pass(cache_vs_z)
           dons_sp = Dict()
           FT = typeof(cache_vs_z[1].z)
           dons_sp["z"] = FT[]
           dons_sp["ρ"] = FT[]
           dons_sp["ρu[1]"] = FT[]
           dons_sp["moisture.ρq_tot"] = FT[]
           dons_sp["turbconv.updraft[1].ρa"] = FT[]
           dons_sp["turbconv.updraft[1].ρaw"] = FT[]
           dons_sp["turbconv.updraft[1].ρaθ_liq"] = FT[]
           dons_sp["turbconv.updraft[1].ρaq_tot"] = FT[]
           dons_sp["turbconv.environment.ρatke"] = FT[]
           dons_sp["turbconv.environment.ρaθ_liq_cv"] = FT[]
           dons_sp["turbconv.environment.ρaq_tot_cv"] = FT[]
           for ca in cache_vs_z
               push!(dons_sp["z"], ca.z)
               push!(dons_sp["ρ"], ca.prog.ρ)
               push!(dons_sp["ρu[1]"], ca.prog.ρu[1])
               push!(dons_sp["moisture.ρq_tot"], ca.prog.moisture.ρq_tot)
               push!(dons_sp["turbconv.updraft[1].ρa"], ca.prog.turbconv.updraft[1].ρa)
               push!(dons_sp["turbconv.updraft[1].ρaw"], ca.prog.turbconv.updraft[1].ρaw)
               push!(dons_sp["turbconv.updraft[1].ρaθ_liq"], ca.prog.turbconv.updraft[1].ρaθ_liq)
               push!(dons_sp["turbconv.updraft[1].ρaq_tot"], ca.prog.turbconv.updraft[1].ρaq_tot)
               push!(dons_sp["turbconv.environment.ρatke"], ca.prog.turbconv.environment.ρatke)
               push!(dons_sp["turbconv.environment.ρaθ_liq_cv"], ca.prog.turbconv.environment.ρaθ_liq_cv)
               push!(dons_sp["turbconv.environment.ρaq_tot_cv"], ca.prog.turbconv.environment.ρaq_tot_cv)
           end
           return dons_sp
       end
dons_single_pass (generic function with 1 method)

julia> let
       #! format: on

       @btime dons_arr = [dons(cache_vs_z) for cache_vs_z in cache_arr]
       @btime dons_arr = [dons_single_pass(cache_vs_z) for cache_vs_z in cache_arr]
       nothing
       end
  29.269 μs (200 allocations: 108.81 KiB)
  515.792 μs (10727 allocations: 441.56 KiB)
```
FWIW, this is only done once at the end of the simulation to compare with our ground truth.